src/plugins/mer/merbuildenginedetailswidget.cpp

```diff
@@ -402,5 +415,11 @@ void MerBuildEngineDetailsWidget::onWwwProxyExcludesEdited(const QString &s)
     emit wwwProxyChanged(m_wwwProxy, m_wwwProxyServerList, m_wwwProxyExcludes);
 }
 
+void MerBuildEngineDetailsWidget::updateSshPortAvailableLabel()
+{
+    const bool available = MerLocalhostAvailablePorts::isAvailable(m_ui->sshPortSpinBox->value());
+    m_ui->sshPortAvailableLabel->setHidden(available || !m_vmOff);
+}
+
```

Comment:
Looking at this again, we are not doing the right thing on the right place. This class should implement just the UI. We should check the port availability in `MerBuildEngineOptionsWidget::onVmOffChanged`:

```
void MerBuildEngineOptionsWidget::onVmOffChanged(bool vmOff)
{
    ...
    m_ui->buildEngineDetailsWidget->setVmOffStatus(vmOff);
    m_ui->buildEngineDetailsWidget->setSshPortOccupied(
            !vmOff && Sfdk::isPortOccupied(emulator->sshPort()));
}
```

```
void MerBuildEngineDetailsWidget::setSshPortOccupied(bool occupied)
{
    m_ui->sshPortOccupiedLabel->setHidden(!occupied);
}
```

`src/libs/sfdk/utils.{h,cpp}` would be a good place for `Sfdk::isPortOccupied` I think.

Reply:
Ok.